detect_secrets/settings.py

```diff
@@ -71,6 +71,7 @@ def default_settings() -> Generator['Settings', None, None]:
             for plugin_type in get_mapping_from_secret_type_to_class().values()
         ],
     }) as settings:
+        get_mapping_from_secret_type_to_class.cache_clear()
```

Comment:
How much does the performance get affected by adding this?

Reply:
`settings.defailt_settings` is called usually only once to setup the parser or run tests, so performance impact should be minimal